CMake/AbseilHelpers.cmake

```diff
@@ -100,9 +111,14 @@ function(absl_cc_library)
 
     if(NOT ABSL_CC_LIB_IS_INTERFACE)
       add_library(${_NAME} STATIC "")
```

Comment:
To support shared libraries (Please ignore, if this is not a desired feature at all):

```suggestion
      add_library(${_NAME})
```

The user can now choose between static and shared libraries via [`-DBUILD_SHARED_LIBS=ON/OFF`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html). For good practice, you may want to explicitely declare this build switch with an explicit default via `option()`. Without declaring it as an option, the default is `OFF` (=> static).

Because there are library dependencies within the `absl` package, it is [recommended](https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling#recommendations) to add relative `RPATH` entries for relocatability. Here is a snippet how we usually add such entries on Linux and MacOS:

```cmake
set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" isSystemDir)
if("${isSystemDir}" STREQUAL "-1")
  if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
    set(CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS} "-Wl,--enable-new-dtags")
    set(CMAKE_SHARED_LINKER_FLAGS ${CMAKE_SHARED_LINKER_FLAGS} "-Wl,--enable-new-dtags")
    set(CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
  elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
    set(CMAKE_INSTALL_RPATH "@loader_path/../${CMAKE_INSTALL_LIBDIR}")
  endif()
endif()
```